src/nagini_translation/translators/obligation/fork.py

```diff
@@ -0,0 +1,226 @@
+import ast
+import copy
+import operator
+
+from typing import List
+
+from nagini_translation.lib import silver_nodes as sil
+from nagini_translation.lib.constants import (
+    GET_ARG_FUNC,
+    GET_METHOD_FUNC,
+    GET_OLD_FUNC,
+    JOINABLE_FUNC,
+    METHOD_ID_DOMAIN,
+    OBJECT_TYPE,
+    THREAD_DOMAIN,
+    THREAD_POST_PRED,
+)
+from nagini_translation.lib.context import Context
+from nagini_translation.lib.program_nodes import PythonMethod
+from nagini_translation.lib.typedefs import (
+    Expr,
+    Info,
+    Position,
+    Stmt
+)
+from nagini_translation.lib.util import (
+    InvalidProgramException,
+    OldExpressionCollector,
+    OldExpressionNormalizer,
+    pprint,
+)
+from nagini_translation.translators.obligation.manager import (
+    ObligationManager,
+)
+from nagini_translation.translators.obligation.node_constructor import (
+    StatementNodeConstructorBase,
+)
+
+
+class ObligationMethodForkConstructor(StatementNodeConstructorBase):
+    """A class that creates a method call node with obligation stuff."""
+
+    def __init__(
+            self, targets, thread: Expr,
+            position: Position, info: Info,
+            translator: 'AbstractTranslator', ctx: Context,
```

Comment:
It means the same thing, it's a type annotation in both cases. The problem is that if you write it not as a string but as a reference to the name, that reference will be evaluated when the file is executed, so the name must be defined, otherwise you get an error. So if AbstractTranslator wasn't written as a string here, I'd have to actually add an import that imports that type.